netlogo-gui/src/main/editor/EditorMenu.scala

```diff
@@ -0,0 +1,9 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.editor
+
+import javax.swing.Action
+
+trait EditorMenu {
+  def offerAction(action: Action): Unit
```

Comment:
In java awt / swing `add` is used on containers to indicate that a component should be added. The `add` method implies two things: (1) that the object will definitely be added after `add` is called and (2) if `add(a)` is called before `add(b)`, `a` will appear "before" `b` within the container. In this case, those implications are false. A menu may or may not add an entry for an action and the order in which the actions are added doesn't influence the order in which they are contained. In light of that, I thought `offer` was a clearer term. Since `offer` doesn't necessarily imply that an action was `add`ed, `revoke` doesn't necessarily imply that it will be removed (that is, it isn't an error to `revoke` an action even if the menu to which it was offered didn't add it during the offer).